src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BitbucketService.java

```diff
@@ -256,7 +236,96 @@ public void createUser(String username, String password, String emailAddress, St
     }
 
     /**
-     * Adds an Bitbucket user to (multiple) Bitbucket groups
+     * Updates an user on Bitbucket
+     *
+     * @param username     The username of the user
+     * @param emailAddress The new email address
+     * @param displayName  The new display name
+     * @throws BitbucketException
+     */
+    public void updateUserDetails(String username, String emailAddress, String displayName) throws BitbucketException {
+        UriComponentsBuilder userDetailsBuilder = UriComponentsBuilder.fromHttpUrl(bitbucketServerUrl + "/rest/api/latest/admin/users");
+        Map<String, Object> userDetailsBody = new HashMap<>();
+        userDetailsBody.put("name", username);
+        userDetailsBody.put("email", emailAddress);
+        userDetailsBody.put("displayName", displayName);
+        HttpEntity<Map<String, Object>> userDetailsEntity = new HttpEntity<>(userDetailsBody, null);
+
+        log.debug("Updating Bitbucket user {} ({})", username, emailAddress);
+
+        try {
+            restTemplate.exchange(userDetailsBuilder.build().encode().toUri(), HttpMethod.PUT, userDetailsEntity, Void.class);
+        }
+        catch (HttpClientErrorException e) {
+            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
+                if (e.getMessage() != null && e.getMessage().contains("com.atlassian.bitbucket.user.NoSuchUserException")) {
```

Comment:
This condition is repeated a few times. Moving it to its own method could reduce duplication.

Reply:
It exists three times. and the log is the same only two times. Not sure if it's really that work, what do your think?

Reply:
I think it is worth it as that way the message has to be only changed in one place in case Bitbucket decides to change the content of `e.getMessage()` in the future. That would make the ‘parsing’ of the error type from the exception message less fragile.

Reducing this in all three places to
```java
if (isUserNotFoundException(e)) {
    …
}
```
would also a lot more readable compared to
```java
if (HttpStatus.NOT_FOUND.equals(e.getStatusCode() && e.getMessage() != null && e.getMessage().contains("com.atlassian.bitbucket.user.NoSuchUserException")) {
    …
}
```
independent of the different contents of the if-block.

Reply:
Gotcha, I didn't realise you only referred to the conditionals. Thought you meant the whole clause.